app/nginx-1.11.10/src/event/ngx_event.c

```diff
@@ -241,6 +253,18 @@ ngx_process_events_and_timers(ngx_cycle_t *cycle)
 
     (void) ngx_process_events(cycle, timer, flags);
 
+#if (NGX_HAVE_FSTACK)
+    /* handle message from kernel (PS: signals from master) in case of network inactivity */
+    if (ngx_current_msec - initial >= ngx_schedule_timeout || tick >= ngx_schedule_timeout) {
+        (void) ngx_ff_process_host_events(cycle, 0, flags);
+
+        /* Update timer*/
+        initial = ngx_current_msec;
+        tick = 0;
+    }
+    tick++;
```

Comment:
There seems to be a problem here, `ngx_schedule_timeout`'s unit is millisecond, but `tick` is not, every main loop iteration, `tick` will plus 1, the comparison `tick >= ngx_schedule_timeout` is unreasonable.

Reply:
Nginx reduces ```gettimeofday``` system calls by caching time variables to improve efficiency. So ```ngx_current_msec``` would not be updated in each loop iteration. So ```tick``` is designed to avoid inability to poll host ,when there is no enent.

I just assume that a loop iteration is equal to one millisecond. Do you mean this is a unreasonable conversion?
I want to modify the comparison to ```tick >= ngx_schedule_timeout * n``` (for one ```n =1000```), Do you agree with this?

Reply:
> I just assume that a loop iteration is equal to one millisecond. Do you mean this is a unreasonable conversion?
I want to modify the comparison to tick >= ngx_schedule_timeout * n (for one n =1000), Do you agree with this?

F-Stack is running on polling mode, the time of one loop iteration is much less than one millisecond, its value is related to the performance of CPU and the number of events in this iteration. It's not fixed.

> Nginx reduces gettimeofday system calls by caching time variables to improve efficiency. And ngx_current_msec would not be updated in each loop iteration. So tick is designed to avoid inability to poll host ,when there is no event.

I consider that `ngx_current_msec` is ok, nginx's related code is below:
```
void
ngx_process_events_and_timers(ngx_cycle_t *cycle)
{
    ngx_uint_t  flags;
    ngx_msec_t  timer, delta;

    if (ngx_timer_resolution) {
        timer = NGX_TIMER_INFINITE;
        flags = 0;

    } else {
        timer = ngx_event_find_timer();
        flags = NGX_UPDATE_TIME;
        ....
   }
   .....
}
```
if `NGX_UPDATE_TIME` is set, `ngx_current_msec` will be updated.

Reply:
I reanalyze the code, and see your point about this.